platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -1154,7 +1155,14 @@ Response handleLogoutRequest(
       return Response.ok("Logout already in progress").build();
     }
 
-    logoutState = new LogoutState(getActiveSps(cookie.getValue()));
+    logoutState =
+        new LogoutState(
+            getActiveSps(cookie.getValue()),
+            logoutRequest
+                .getSessionIndexes()
+                .stream()
+                .map(SessionIndex::getSessionIndex)
```

Comment:
❓ Do we need to filter out any `null` or empty session indices? If not, why not just do `new ArrayList(logoutRequest.getSessionIndexes())` instead of using a stream?

Reply:
It might actually be best to just move the defensive copying logic into the constructor such that its not the callers responsibility.  Once in there we could simply do a `ImmutableList.copyOf(..)` or similiar.

Reply:
Given how `LogoutState` is initialized, it should probably have a constructor that accepts a `LogoutRequest`. But that's more refactoring than I want to attempt right now.

The SAML library wraps the actual session index (string) inside of an object. The `map` in the streaming expression to to extract the `String` from the `SessionIndex` object. 

I reviewed where the constructor was used, and decided the best course of action was to add a setter, `setSessionIndexObjects`, that extracts the strings.